Refactor: Clean Up License

jens-erdmann opened the pull request:
Clean up license.py by removing debugging code and simplifying code.
See commits for details.

jens-erdmann committed:
License: Remove Dead Code

Remove code in license.py that is disabled and was used to debug.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
License: Remove Excessive Logging

Remove logging needed for debugging and is not really adding clarity to
the code.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
License: Simplify read_symbols

Simplify read_symbols by using list comprehension.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
License: Simplify translate_and_relicense

Remove unneeded string replaccements in translate_and_relicense as it is
already done in translate method.
Remove unneeded checks for None and make code more pythonic.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
License: Rename Method Parameter

Rename method parameter in interim_license_expression_list as it
contains an underscore and should not be used here.
Also change the name of the variable that was named the same to
somehting more representative.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
License: Simplify License Expression List Methods

Simplify license expression list method and its helper methods by
removing nesting, using f-strings and reducing duplication.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

priv-kweihmann commented on flict/flictlib/license.py:
```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
-                    logger.license_logger.debug("    list: " + str(new_list))
-                license_list = new_list
-            return license_list
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
```
Shouldn't that be a FlictError instead?
I honestly don't want to introduce plain Exceptions where it's not needed

jens-erdmann replied:
Good point. Was thinking about that myself when I crossed this.

jens-erdmann replied:
Fixed

priv-kweihmann commented on flict/flictlib/license.py:
```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
-                    logger.license_logger.debug(
-                        "    item: " + str(license_list) + " <--- " + str(lep_item))
-                    new_list.append(lep_item)
+            raise Exception(f"Internal failure. Wrong type {lep} for: {lep}")
```
Same here

jens-erdmann replied:
Fixed

priv-kweihmann commented on flict/flictlib/license.py:
```diff
@@ -340,91 +293,54 @@ def interim_license_expression_set_list(self, interim_license_expression_list):
+        if len(license_list) == 0:
+            return lep
+
+        new_list = license_list
```
I guess a `return list(lep)` could be used for this whole block here

jens-erdmann replied:
But wouldn't that miss out on the potential content of `license_list`?

jens-erdmann replied:
I gave it a spin and it fails a bunch of tests.

priv-kweihmann replied:
True so it is `return list(license_list) + list(lep)` then - maybe also worth making it unique by doing `return list(set(list(license_list) + list(lep)))`

jens-erdmann replied:
Just more errors.

priv-kweihmann replied:
in this case - leave it for now as it is

jens-erdmann committed:
Return Codes: Add Code for Internal Error

Add a new return code for internal errors that are raised due to
unexpected behavior.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>

jens-erdmann committed:
License: Replace Exceptions with FlictError

Replace python builtin exceptions with FlictErrors to properly identify
them and handle them accordingly.
Plain exceptions are too generic and might lead to overcompensation.

Signed-off-by: Jens Erdmann <jens.erdmann@web.de>
